pipeline_tools/create_analysis_metadata.py

```diff
@@ -0,0 +1,502 @@
+#!/usr/bin/env python
+import argparse
+import base64
+import codecs
+import json
+from copy import deepcopy
+from csv import DictReader
+from google.cloud import storage
+from typing import List
+
+
+def create_analysis_process(raw_schema_url,
+                            metadata_file,
+                            analysis_process_schema_version,
+                            analysis_id,
+                            input_bundles_string,
+                            reference_bundle,
+                            inputs,
+                            output_url_to_md5,
+                            extension_to_format,
+                            analysis_file_version,
+                            run_type):
+    """Collect and create the information about the analysis process for submission to Ingest service.
+
+    Based the design of this HCA metadata type, analysis_process will vary between each analysis run, even if they share
+    the same version of pipeline. Most of the content of this process will be collected from the Cromwell workflow
+    metadata. This function strictly follows the metadata schema defined at:
+    https://github.com/HumanCellAtlas/metadata-schema
```

Comment:
This should point to the json schema the function is using, although it's only on the integration branch at the moment: https://github.com/HumanCellAtlas/metadata-schema/blob/integration/json_schema/type/process/analysis/analysis_process.json

Reply:
will do!